api/http-api.md

```diff
@@ -0,0 +1,177 @@
+  * `RespondActivityTaskCompletedById` - `POST /api/v1/namespaces/{namespace}/activities/complete-by-id`
+  * `RespondActivityTaskFailed` - `POST /api/v1/namespaces/{namespace}/activities/fail`
+  * `RespondActivityTaskFailedById` - `POST /api/v1/namespaces/{namespace}/activities/fail-by-id`
+  * `RespondActivityTaskCanceled` - `POST /api/v1/namespaces/{namespace}/activities/cancel`
+  * `RespondActivityTaskCanceledById` - `POST /api/v1/namespaces/{namespace}/activities/cancel-by-id`
+* Search attributes:
+  * Intentionally not exposed high-level search attribute calls from workflow service
+* Task polling/completing:
+  * Intentionally not exposed (except for activity task stuff needed for async activities)
+* Task queues:
+  * `ResetStickyTaskQueue` - Intentionally not exposed
+  * `DescribeTaskQueue` - `GET /api/v1/namespaces/{namespace}/task-queues/{task_queue_name}`
+  * `ListTaskQueuePartitions` - Intentionally not exposed
+* Schedules:
+  * `CreateSchedule` - `POST /api/v1/namespaces/{namespace}/schedules/{schedule_id}`
+  * `DescribeSchedule` - `GET /api/v1/namespaces/{namespace}/schedules/{schedule_id}`
+  * `UpdateSchedule` - `POST /api/v1/namespaces/{namespace}/schedules/{schedule_id}/update`
+    * This is not a `PUT` because update different from create and not `PATCH` to avoid confusion with below
+  * `PatchSchedule` - `POST /api/v1/namespaces/{namespace}/schedules/{schedule_id}/patch`
+    * This is not a `PATCH` to avoid confusion with update above
+  * `ListScheduleMatchingTimes` - `GET /api/v1/namespaces/{namespace}/schedules/{schedule_id}/matching-times`
+  * `DeleteSchedule` - `DELETE /api/v1/namespaces/{namespace}/schedules/{schedule_id}`
+    * Confirm whether eventually consistent
+  * `ListSchedules` - `GET /api/v1/namespaces/{namespace}/schedules`
+* Batch operations:
+  * `StartBatchOperation` - `POST /api/v1/namespaces/{namespace}/batch-operations/{job_id}`
+  * `DescribeBatchOperation` - `GET /api/v1/namespaces/{namespace}/batch-operations/{job_id}`
+  * `ListBatchOperations` - `GET /api/v1/namespaces/{namespace}/batch-operations`
+  * `StopBatchOperation` - `POST /api/v1/namespaces/{namespace}/batch-operations/{job_id}/stop`
+* Worker build IDs:
+  * Intentionally not exposed until they settle
+* Other calls:
+  * `GetClusterInfo` - Intentionally not exposed
+  * `GetSystemInfo` - `GET /api/v1/system-info`
+
+### Embedding into Server
+
+Current suggested approach, subject to discussion:
+
+* Generated gRPC gateway code will be auto-generated at the same time as api-go is for gRPC and live alongside it
+* Service config has new `httpPort`
+  * Only allowed on "frontend" service, presence elsewhere is config failure
+  * Default value is `grpcPort` + 10
+    * Makes `7243` the default - unassigned in IANA, unused in server
+    * Cannot think of case where +10 clashes, e.g. if they use 80 or 443 or 8080 or whatever, 90, 453, and 8090 are fine
+* Service config has new `disableHttp`
```

Comment:
Maybe this clashes with existing config conventions, but I don't generally like having negatives in variable names, because setting something like `disableHttp=true` might parse strangely especially for non-native English speakers.

I suggest naming this `enableHttp`, and have the default be `true`. Then if a user wants to disable HTTP, they can set `enableHttp=false` explicitly.

Reply:
hrm. If we're already consistently using `disableX` names for config values, then yes, much as I hate to say it, it's probably better to be consistently confusing than inconsistently confusing (which just adds to the confusion).